engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowFinalizationActor.scala

```diff
@@ -58,21 +64,36 @@ case class WorkflowFinalizationActor(workflowId: WorkflowId, workflowDescriptor:
       val backendFinalizationActors = Try {
         for {
           (backend, calls) <- workflowDescriptor.backendAssignments.groupBy(_._2).mapValues(_.keys.toSeq)
-          props <- CromwellBackends.shadowBackendLifecycleFactory(backend).map(_.workflowFinalizationActorProps(workflowDescriptor.backendDescriptor, calls)).get
+          props <- CromwellBackends.shadowBackendLifecycleFactory(backend).map(
+            _.workflowFinalizationActorProps(workflowDescriptor.backendDescriptor, calls,
+              executionStore.toBackendExecutionStore, outputStore.toBackendOutputStore)).get
           actor = context.actorOf(props)
-        } yield (actor, backend)
+        } yield actor
+      }
+
+      val engineFinalizationActors = Try {
+        val props = Seq(
+          CopyWorkflowLogsActor.props(workflowId, workflowDescriptor, executionStore, outputStore),
+          CopyWorkflowOutputsActor.props(workflowId, workflowDescriptor, executionStore, outputStore))
+        props map context.actorOf
       }
 
-      backendFinalizationActors match {
+      val allActors = for {
+        backendFinalizationActorsFromTry <- backendFinalizationActors
+        engineFinalizationActorsFromTry <- engineFinalizationActors
+      } yield backendFinalizationActorsFromTry ++ engineFinalizationActorsFromTry
+
+      allActors match {
         case Failure(ex) =>
           sender ! WorkflowFinalizationFailedResponse(Seq(ex))
           goto(WorkflowFinalizationFailedState)
         case Success(actors) if actors.isEmpty =>
           sender ! WorkflowFinalizationSucceededResponse
           goto(FinalizationSucceededState)
         case Success(actors) =>
-          actors.keys.foreach(_ ! Finalize)
-          goto(FinalizationInProgressState) using stateData.withBackendActors(actors)
+          val actorSet = actors.toSet
```

Comment:
Found and fixed. It was that scatters are in the store as calls with and without indexes.